common/containers/Tabs/GenerateWallet/components/Mnemonic/Mnemonic.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import { generateMnemonic } from 'bip39';
+import translate from 'translations';
+import Word from './Word';
+import FinalSteps from '../FinalSteps';
+import { WalletType } from '../../GenerateWallet';
+import './Mnemonic.scss';
+
+interface State {
+  words: string[];
+  confirmValues: string[];
+  isConfirming: boolean;
+  isConfirmed: boolean;
+}
+
+interface WordTuple {
+  word: string;
+  index: number;
+}
+
+export default class GenerateMnemonic extends React.Component<{}, State> {
+  public state: State = {
+    words: [],
+    confirmValues: [],
+    isConfirming: false,
+    isConfirmed: false
+  };
+
+  public componentDidMount() {
+    this.regenerateWordArray();
+  }
+
+  public render() {
+    const { words, confirmValues, isConfirming, isConfirmed } = this.state;
+    let content;
+
+    if (isConfirmed) {
+      content = <FinalSteps walletType={WalletType.Mnemonic} />;
+    } else {
+      const canContinue = this.checkCanContinue();
+      const firstHalf: WordTuple[] = [];
+      const lastHalf: WordTuple[] = [];
+      words.forEach((word, index) => {
+        if (index < words.length / 2) {
+          firstHalf.push({ word, index });
+        } else {
+          lastHalf.push({ word, index });
+        }
+      });
+
+      content = (
+        <div>
+          <h1 className="GenerateMnemonic-title">Generate a {translate('x_Mnemonic')}</h1>
+
+          <p className="GenerateMnemonic-help">
+            {isConfirming
+              ? `
+              Re-enter your phrase to confirm you copied it correctly. If you
+              forgot one of your words, just click the button beside the input
+              to reveal it.
+            `
+              : `
+              Write these words down. Do not copy them to your clipboard, or save
+              them anywhere online.
+            `}
+          </p>
+
+          <div className="GenerateMnemonic-words">
+            {[firstHalf, lastHalf].map((ws, i) => (
```

Comment:
I would consider unrolling the first level array just to reduce complexity, with having a helper function / SFC generate the `Word` components

Reply:
Can you elaborate on
> unrolling the first level array
Not sure what that looks like

Reply:
I think I simplified this by making a makeWord helper function.